common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -1154,6 +1154,14 @@ private void startThread() {
                     }
                 }
             }
+        } else if (currentState == ST_SUSPENDING) {
+            while (state == ST_SUSPENDING) {
+                if (!STATE_UPDATER.compareAndSet(this, currentState, ST_STARTED)) {
+                    continue;
```

Comment:
Should I check here in the CAS loop if the state goes from suspending to suspended while attempting to start it and if so, attempt to do the start logic?